src/components/editor/EditAuthorList.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import _ from 'lodash';
+import { shallowEquals } from 'lib/BaseComponent';
+
+export default class EditAuthorList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // Key is slug, value is name
+      authorsAdded: {},
+      // An object where key is slug and value is whether it's deleted or not
+      authorsDeleted: this.props.data.reduce((cur, next) => {
+        cur[next.slug] = false;
+        return cur;
+      }, {})
+    };
+    this.handleAddAuthor = this.handleAddAuthor.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
And I was just a bit suspicious about the shallowEquals part when it checks ObjA[key] === ObjB[key] as I checked and this checks if it references the same object. So I was just thinking that if it just checks authorsAdded === newAuthorsAdded that might do something false some times?